assembly/src/ast/program.rs

```diff
@@ -327,3 +327,37 @@ impl fmt::Display for ProgramAst {
         writeln!(f, "end")
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for ProgramAst {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.body.write_into(target);
+        target.write_u64(self.local_procs.len() as u64);
+        self.local_procs.write_into(target);
+        self.import_info.write_into(target);
+        self.start.write_into(target);
+    }
+}
+
+impl Deserializable for ProgramAst {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let body = CodeBody::read_from(source)?;
+
+        let count = source
+            .read_u64()?
+            .try_into()
+            .map_err(|v| DeserializationError::InvalidValue(format!("can't fit {v} into usize")))?;
+        let local_procs = ProcedureAst::read_batch_from(source, count)?;
```

Comment:
minor nit: I'd prefer a more descriptive naming - e.g., `num_local_procs` instead of `count`.

Reply:
changed